src/util-thash.c

```diff
@@ -204,6 +204,19 @@ static void THashDataFree(THashTableContext *ctx, THashData *h)
 #define GET_VAR(prefix,name) \
     snprintf(varname, sizeof(varname), "%s.%s", (prefix), (name))
 
+static void THashConfigValidate(const char *confvalue, char *varname)
+{
+  int varlen = strlen(confvalue);
```

Comment:
I was thinking ``ByteExtractStringUint32`` would do this for us, but it quite explicitly does not:
```c
    /* This will interfere with some rules that do not know the length
     * in advance and instead are just using the max.
     */
#if 0
    else if (len && *endptr != '\0') {
        fprintf(stderr, "ByteExtractString: Extra characters following numeric value\n");
        return -1;
    }
#endif
```

I guess I would like a more generic solution though, but I fear the risk of side effects if we enable it for ``ByteExtractStringUint32``.

cc: @shivan1b

Reply:
Ah, so are there other places where this sort of validation could take place where it currently doesn't? I am guessing by your comment there is :)
I was also wondering if it would be worth adding a max value check to ensure values aren't larger than their type?

Reply:
The functions are per type, so that part is mostly covered. Though in some cases the type allows for much larger values than make sense. For example the ``hash-size`` is a ``uint32_t``, but only because ``uint16_t`` is too small, not because we need the full 32 bits.

I did check ``ByteExtractString`` for enabling the 'trailing chars' check, but it can't be done in its current form. These functions are not just used by parsing rules or config, but also to extract data from packets for ``byte_jump`` and ``byte_test`` and maybe in other places. So for parsing we might need a new set of functions to do what we want.

Reply:
thanks for the additional information!